test/jdk/java/awt/print/PrinterJob/PrintAllFonts.java

```diff
@@ -79,18 +87,20 @@ public int print(Graphics g, PageFormat pf, int pageIndex) {
         } else {
             fontNum = startNum;
         }
-        g.setColor(Color.black);
 
-        int hgt = (int)pf.getImageableHeight();
-        int fontsPerPage = hgt/lineHeight;
-        int x = (int)pf.getImageableX()+10;
-        int y = (int)pf.getImageableY()+lineHeight;
+        int fontsPerPage = (int) pf.getImageableHeight() / lineHeight;
```

Comment:
```suggestion
        int fontsPerPage = (int) pf.getImageableHeight() / lineHeight - 1;
```
Allow for some margin at the end of the page, otherwise most fonts printed at the bottom of a page are clipped.